rclpy/rclpy/node.py

```diff
@@ -30,13 +33,35 @@
 
 class Node:
 
-    def __init__(self, handle):
+    def __init__(self, node_name, *, namespace=None):
         self.clients = []
-        self._handle = handle
+        self._handle = None
         self.publishers = []
         self.services = []
         self.subscriptions = []
         self.timers = []
+        self._default_callback_group = MutuallyExclusiveCallbackGroup()
+
+        namespace = namespace or ''
+        failed = False
+        if not ok():
+            raise NotInitializedException('cannot create node')
+        try:
+            node_handle = _rclpy.rclpy_create_node(node_name, namespace)
+            self._handle = node_handle
+        except ValueError:
+            failed = True
+        if failed:
```

Comment:
The variable `failed` seems to be unnecessary. The error handling code could just be in the `except` block.

Reply:
Removed in 3f66415d4788c4c0d6c06e4529c17b13640e9948